backend/driver.py

```diff
@@ -114,6 +114,7 @@ def _generate_launcher(constants, signature, kernel_name):
 static void _launch(int gridX, int gridY, int gridZ, {arg_decls}) {{
   if (gridX*gridY*gridZ > 0) {{
     // Cast "function" to the real function type.
+    {"#pragma omp parallel for collapse(3)" if _get_sanitizer_type() == "tsan" else ""}
```

Comment:
can we add a comment here explaining the reasoning we need to parallelize when using with tsan